### PR TITLE
Implement add delete in use decks client

### PR DESCRIPTION
**Pull request by derricklee43**
- implemented add and delete deck with the associated routing
- delete endpoint is a little scuffed, but we can figure it out after the demo

### CODE REVIEW

File: src/pages/edit-deck-page/edit-deck-page.tsx [hunk 1]
```diff
@@ -53,14 +53,20 @@ export const EditDeckPage = () => {
...
+  async function handleCreateDeckClick(deck: Deck) {
+    const addedDeck = await addDeck(deck);
+    navigate(`${paths.deck}/${addedDeck.metaData.id}`);
   }
 
-  function handleDeleteDeckClick() {
-    // Todo: delete deck
+  async function handleDeleteDeckClick() {
+    if (!isNewDeck && deck !== undefined) {
+      // Todo: always throws. It looks like the response is bad but the deck is still deleted
```
Comment (jiejasonliu) [inline src/pages/edit-deck-page/edit-deck-page.tsx:63]: oh! i'm 99.99% certain this is on me, delete requests don't have a response so this line would definitely throw:

`const data = response.json()` in `use-fetch-wrapper.ts:39`

you can fix this by changing it to:

```ts
const text = await response.text();
const data = text && JSON.parse(text);
```

**Review by jiejasonliu — APPROVED** [review]
i'm okay with merging this despite API changes coming in the future

we'll just need to be careful for that

File: src/pages/edit-deck-page/edit-deck-page.tsx (hunk 1 above)
Comment (derricklee43) [reply src/pages/edit-deck-page/edit-deck-page.tsx]: sweet, that fixed it

**Comment by derricklee43** [comment]
Yeah, we can just leave it open until the API is fixed and we have it updated

**Comment by derricklee43** [comment]
Ok, we should be good to go for these updated endpoints

**Review by jiejasonliu — APPROVED** [review]
LGTM, do you mind updating the comments w the correct endpoint info?

**Comment by derricklee43** [comment]
yep, there are now updated

**Pull request merged**